src/control/cmd/daos_server/network.go

```diff
@@ -91,3 +66,41 @@ func fabricInterfaceSetToHostFabric(fis *hardware.FabricInterfaceSet, filterProv
 
 	return hf
 }
+
+func GetLocalHostFabric(ctx context.Context, fs *hardware.FabricScanner, filterProvider string) (*control.HostFabric, error) {
+	results, err := fs.Scan(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return fabricInterfaceSetToHostFabric(results, filterProvider), nil
+}
+
+func (cmd *networkScanCmd) Execute(_ []string) error {
+	ctx := context.Background()
+	fs := hwprov.DefaultFabricScanner(cmd.Logger)
+
+	if cmd.FabricProvider == "" {
+		cmd.FabricProvider = cmd.config.Fabric.Provider
+	}
+	if cmd.FabricProvider == "" {
+		cmd.FabricProvider = allProviders
+	}
```

Comment:
The overall PR is LGTM, just this point which probably be fixed before merge.

Reply:
I don't see a problem with the current code, the value is only set to `allProviders` if `cmd.FabricProvider` and `cmd.config.Fabric.Provider` are empty. In your suggestion you overwrite any value provided in `cmd.FabricProvider`.

Reply:
My apologies, I have misunderstood the code.
The two sequential tests of the same value confused me.

Reply:
I think this is a valid code critique. It is a bit confusing to reuse `cmd.FabricProvider` this way, and I think it could lead to errors later when the code is modified by someone else. How about something like this:

```go
// Default to including all providers in scan, but allow overrides by the CLI option or configuration.
fabricProvider := allProviders
if cmd.FabricProvider == "" && cmd.config.Fabric.Provider != "" {
        fabricProvider = cmd.config.FabricProvider
} else if cmd.FabricProvider != "" {
        fabricProvider = cmd.FabricProvider
}
```